plugins/notifications-backend-module-email/src/processor/NotificationsEmailProcessor.ts

```diff
@@ -215,20 +217,45 @@ export class NotificationsEmailProcessor implements NotificationProcessor {
     );
   }
 
-  private async sendPlainEmail(notification: Notification, emails: string[]) {
+  private getNotificationLink(notification: Notification) {
+    if (notification.payload.link) {
+      const isRelativeUrl =
+        !notification.payload.link.match(/^([a-z]*:)?\/\//i);
+      if (isRelativeUrl) {
+        const url = new URL(this.frontendBaseUrl);
+        url.pathname = notification.payload.link;
```

Comment:
True, that constructor should work better. I will change!

I just hope there was backend version of the `useRouteRef` because it's a bit hard to make these links now for notifications that point somewhere inside the backstage app. But let's discuss that maybe later.